### PR TITLE
Add TemplateRegistryAwareInterface

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

I am targeting this branch, because this change respect BC.

## Changelog

```markdown
### Added
-  `Sonata\AdminBundle\Templating\TemplateRegistryAwareInterface` with `getTemplateRegistry()`, `hasTemplateRegistry()` and `setTemplateRegistry()` methods
```

### CODE REVIEW

File: src/Admin/AbstractAdmin.php
Comment (OskarStark) [inline src/Admin/AbstractAdmin.php:3268]: Why it can return null?

File: src/Templating/TemplateRegistryOwnerInterface.php
Comment (OskarStark) [inline src/Templating/TemplateRegistryOwnerInterface.php:27]: And here it cannot?

File: src/Admin/AbstractAdmin.php
Comment (OskarStark) [inline src/Admin/AbstractAdmin.php:1356]: Imo this is a BC break

File: src/Admin/AbstractAdmin.php
Comment (VincentLanglet) [reply src/Admin/AbstractAdmin.php:3268]: I can in 3.x.

It won't in master. A next major would be needed indeed.

File: src/Admin/AbstractAdmin.php
Comment (VincentLanglet) [reply src/Admin/AbstractAdmin.php:1356]: it's final

**Comment by phansys** [comment]
Other packages use the term "aware" instead "owner" for cases like this (`ContainerAwareInterface`, `LoggerAwareInterface`, etc).
IMO, based on that premise, we should rename this interface to `TemplateRegistryAwareInterface`.

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
@wbloszyk Please wait for multiple approval before merging (generally 2 is fine).
And if possible, it's generally better to not merge our own PR. ;)

**Comment by wbloszyk** [comment]
I ask about merge on slack. I prefer 1 approve per refactoring PR and 2+ for new feature.

File: src/Templating/TemplateRegistryAwareInterface.php
Comment (VincentLanglet) [inline src/Templating/TemplateRegistryAwareInterface.php:17]: Typo in comment

**Comment by VincentLanglet** [comment]
> I ask about merge on slack. I prefer 1 approve per refactoring PR and 2+ for new feature.

It's a team project. And people can't be connected every day.

**Comment by greg0ire** [comment]
Also, it's open-source software… there are a loooooot more users than a company project.

**Pull request merged**
